cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java

```diff
@@ -202,4 +210,10 @@ private ByteBuffer encodeTaskRequest(RunnableTaskRequest request) throws IOExcep
 
     throw new IllegalArgumentException("Unsupported content encoding " + encoding);
   }
+
+  // Use task worker for executing user code, and system worker for executing system code.
```

Comment:
Use task worker for executing an unsecured logic (e.g., deploying a pipeline which contains user code) remotely on a task worker pod. Use system worker pod for executing secure logic remotely on a system worker pod.